### PR TITLE
planner: update the doc for merge hint

### PR DESCRIPTION
**Pull request by dayicklp**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dayicklp** [comment]
@winoros PTAL again

File: optimizer-hints.md [hunk 1]
```diff
@@ -286,6 +286,16 @@ SHOW WARNINGS;
...
 
+### MERGE()
+
+`MERGE()` 会关闭在执行[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中对当前子查询的物化过程，并将内部查询内联展开到外部查询。该Hint适用于非递归的公共表表达式查询，某些场景下，会比默认开辟一块临时空间的效果更好。例如将外部查询的条件下推： 
+
+{{< copyable "sql" >}}
+
+```sql
+WITH CTE AS (SELECT /*+ MERGE() */ * FROM tc WHERE tc.a < 60) SELECT * FROM CTE WHERE CTE.a <18;
+```
```
Comment (winoros) [inline optimizer-hints.md:297]: 这里在多写一下具体的生效范围吧，必须是简单的 CTE 才可以之类的

**Comment by dayicklp** [comment]
PTAL @winoros

File: optimizer-hints.md [hunk 2]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
+
+`MERGE()` 会关闭在执行[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中对当前子查询的物化过程，并将内部查询内联展开到外部查询。该 Hint 适用于非递归的公共表表达式查询，某些场景下，会比默认开辟一块临时空间的效果更好。例如将外部查询的条件下推以及嵌套的 CTE 查询： 
+
+{{< copyable "sql" >}}
+
+```sql
+-- 利用 Hint 将外部查询条件谓词下推
+WITH CTE AS (SELECT /*+ MERGE() */ * FROM tc WHERE tc.a < 60) SELECT * FROM CTE WHERE CTE.a <18;
+-- 在嵌套 CTE 查询中应用该 Hint 来指定某个 CTE 内联展开到外部查询
+with cte1 as (select * from tc), cte2 as (with cte3 as (select /*+ MERGE() */ * from te) ,cte4 as (select * from tc) select * from cte3,cte4) select * from cte2;
```
Comment (winoros) [inline optimizer-hints.md:299]: 这个 SQL 是不是写错了？

Comment (dayicklp) [reply optimizer-hints.md:299]: 第二个是两层嵌套CTE在cte3应用merge，cte4依旧是默认物化的方式，cte1没有引用

File: develop/dev-guide-use-common-table-expression.md [hunk 3]
```diff
@@ -173,6 +173,10 @@ FROM
 
 值得注意的是，`books_authored_by_rm` 中的查询只会执行一次，TiDB 会开辟一块临时空间对查询的结果进行缓存，当 `books_with_average_ratings` 和 `books_with_orders` 引用时会直接从该临时空间当中获取数据。
 
+> **注意：**
```
Comment (TomShawn) [inline develop/dev-guide-use-common-table-expression.md:176]:
```suggestion
> **建议：**
```

File: develop/dev-guide-use-common-table-expression.md [hunk 4]
```diff
@@ -173,6 +173,10 @@ FROM
...
+>
+> - 当默认执行的CTE效率不高情况下，我们可以选用 [MERGE()](/optimizer-hints.md#merge) hint 的方式，将CTE子查询拓展到外部查询，借此来提高执行效率。
```
Comment (TomShawn) [inline develop/dev-guide-use-common-table-expression.md:178]:
```suggestion
> - 当默认的 CTE 查询执行效率不高时，你可以使用 [`MERGE()`](/optimizer-hints.md#merge) hint，将 CTE 子查询拓展到外部查询，以此提高执行效率。
```

File: optimizer-hints.md (hunk 2 above)
Comment (TomShawn) [inline optimizer-hints.md:299]: 建议 SQL 语句关键字都大写，与上一行保持一致。

File: optimizer-hints.md [hunk 5]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
 +---------+------+-------------------------------------------------------------------------------------------------------------------+
 ```
 
+### MERGE()
+
+`MERGE()` 会关闭在执行[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中对当前子查询的物化过程，并将内部查询内联展开到外部查询。该 Hint 适用于非递归的公共表表达式查询，某些场景下，会比默认开辟一块临时空间的效果更好。例如将外部查询的条件下推以及嵌套的 CTE 查询：
```
Comment (TomShawn) [inline optimizer-hints.md:291]:
```suggestion
在含有[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中使用 `MERGE()` hint，可关闭对当前子查询的物化过程，并将内部查询的内联展开到外部查询。该 hint 适用于非递归的公共表表达式查询，在某些场景下，使用该 hint 会比默认开辟一块临时空间的语句执行效率更高。例如将外部查询的条件下推以及嵌套的 CTE 查询： 
```

File: optimizer-hints.md [hunk 6]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
 ```
 
+### MERGE()
+
+`MERGE()` 会关闭在执行[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中对当前子查询的物化过程，并将内部查询内联展开到外部查询。该 Hint 适用于非递归的公共表表达式查询，某些场景下，会比默认开辟一块临时空间的效果更好。例如将外部查询的条件下推以及嵌套的 CTE 查询： 
+
+{{< copyable "sql" >}}
+
+```sql
+-- 利用 Hint 将外部查询条件谓词下推
```
Comment (TomShawn) [inline optimizer-hints.md:296]:
```suggestion
-- 使用 hint 将外部查询条件的谓词下推
```

File: optimizer-hints.md [hunk 7]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
+WITH CTE AS (SELECT /*+ MERGE() */ * FROM tc WHERE tc.a < 60) SELECT * FROM CTE WHERE CTE.a <18;
+-- 在嵌套 CTE 查询中应用该 Hint 来指定某个 CTE 内联展开到外部查询
```
Comment (TomShawn) [inline optimizer-hints.md:298]:
```suggestion
-- 在嵌套 CTE 查询中使用该 hint 来指定将某个 CTE 内联展开到外部查询
```

File: optimizer-hints.md [hunk 8]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
+with cte1 as (select * from tc), cte2 as (with cte3 as (select /*+ MERGE() */ * from te) ,cte4 as (select * from tc) select * from cte3,cte4) select * from cte2;
+```
+
+> **注意：**
+>
+> `MERGE()` 只适用于简单的 CTE 查询，在某些场景下无法使用该 Hint:
```
Comment (TomShawn) [inline optimizer-hints.md:304]:
```suggestion
> `MERGE()` 只适用于简单的 CTE 查询，在以下情况中无法使用该 hint：
```

File: optimizer-hints.md [hunk 9]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
+> 
+> - 对于[递归的CTE](/develop/dev-guide-use-common-table-expression.md#递归的-cte)
```
Comment (TomShawn) [inline optimizer-hints.md:306]:
```suggestion
> - [递归的 CTE 查询](/develop/dev-guide-use-common-table-expression.md#递归的-cte)
```

File: optimizer-hints.md [hunk 10]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
+> - 子查询中有无法进行内联展开的部分，例如聚合算子、窗口函数以及`DINSTINCT`等。
+> 
+> 特别地，当 CTE 引用次数过多时，性能可能不如默认物化的方式。
```
Comment (TomShawn) [inline optimizer-hints.md:309]:
```suggestion
> - 子查询中有无法进行内联展开的部分，例如聚合算子、窗口函数以及 `DINSTINCT` 等。
> 
> 当 CTE 引用次数过多时，查询性能可能低于默认的物化方式。
```

File: develop/dev-guide-use-common-table-expression.md [hunk 11]
```diff
@@ -173,6 +173,10 @@ FROM
 
 值得注意的是，`books_authored_by_rm` 中的查询只会执行一次，TiDB 会开辟一块临时空间对查询的结果进行缓存，当 `books_with_average_ratings` 和 `books_with_orders` 引用时会直接从该临时空间当中获取数据。
 
+> **建议：**
+>
+> - 当默认的 CTE 查询执行效率不高时，你可以使用 [`MERGE()`](/optimizer-hints.md#merge) hint，将 CTE 子查询拓展到外部查询，以此提高执行效率。
```
Comment (TomShawn) [inline develop/dev-guide-use-common-table-expression.md:178]:
```suggestion
> 当默认的 CTE 查询执行效率不高时，你可以使用 [`MERGE()`](/optimizer-hints.md#merge) hint，将 CTE 子查询拓展到外部查询，以此提高执行效率。
```

File: optimizer-hints.md [hunk 12]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
 
+### MERGE()
+
+在含有[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中使用 `MERGE()` hint，可关闭对当前子查询的物化过程，并将内部查询的内联展开到外部查询。该 hint 适用于非递归的公共表表达式查询，在某些场景下，使用该 hint 会比默认开辟一块临时空间的语句执行效率更高。例如将外部查询的条件下推以及嵌套的 CTE 查询： 
+
+{{< copyable "sql" >}}
+
+```sql
+-- 使用 hint 将外部查询条件的谓词下推
+WITH CTE AS (SELECT /*+ MERGE() */ * FROM tc WHERE tc.a < 60) SELECT * FROM CTE WHERE CTE.a <18;
```
Comment (TomShawn) [inline optimizer-hints.md:297]:
```suggestion
WITH CTE AS (SELECT /*+ MERGE() */ * FROM tc WHERE tc.a < 60) SELECT * FROM CTE WHERE CTE.a <18;

```

File: optimizer-hints.md [hunk 13]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
+-- 在嵌套 CTE 查询中使用该 hint 来指定将某个 CTE 内联展开到外部查询
+WITH CTE1 AS (SELECT * FROM t1), CTE2 AS (WITH CTE3 AS (SELECT /*+ MERGE() */ * FROM t2) ,CTE4 AS (SELECT * FROM t3) SELECT * FROM CTE3,CTE4) SELECT * FROM CTE1,CTE2;
+```
+
+> **注意：**
+>
+> `MERGE()` 只适用于简单的 CTE 查询，在以下情况中无法使用该 hint：
```
Comment (TomShawn) [inline optimizer-hints.md:304]:
```suggestion
> `MERGE()` 只适用于简单的 CTE 查询，在以下情况下无法使用该 hint：
```

File: optimizer-hints.md [hunk 14]
```diff
@@ -286,6 +286,28 @@ SHOW WARNINGS;
...
+> 
+> - [递归的 CTE 查询](/develop/dev-guide-use-common-table-expression.md#递归的-cte)
+> - 子查询中有无法进行内联展开的部分，例如聚合算子、窗口函数以及 `DINSTINCT` 等。
```
Comment (TomShawn) [inline optimizer-hints.md:307]:
```suggestion
> - 子查询中有无法进行内联展开的部分，例如聚合算子、窗口函数以及 `DINSTINCT` 等
```

File: optimizer-hints.md [hunk 15]
```diff
@@ -286,6 +286,29 @@ SHOW WARNINGS;
 +---------+------+-------------------------------------------------------------------------------------------------------------------+
 ```
 
+### MERGE()
+
+在含有[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中使用 `MERGE()` hint，可关闭对当前子查询的物化过程，并将内部查询的内联展开到外部查询。该 hint 适用于非递归的公共表表达式查询，在某些场景下，使用该 hint 会比默认开辟一块临时空间的语句执行效率更高。例如将外部查询的条件下推以及嵌套的 CTE 查询：
```
Comment (TomShawn) [inline optimizer-hints.md:291]:
```suggestion
在含有[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中使用 `MERGE()` hint，可关闭对当前子查询的物化过程，并将内部查询的内联展开到外部查询。该 hint 适用于非递归的公共表表达式查询，在某些场景下，使用该 hint 会比默认分配一块临时空间的语句执行效率更高。例如将外部查询的条件下推或在嵌套的 CTE 查询中： 
```

File: optimizer-hints.md [hunk 16]
```diff
@@ -286,6 +286,29 @@ SHOW WARNINGS;
...
 
+### MERGE()
+
+在含有[公共表表达式](/develop/dev-guide-use-common-table-expression.md)的查询中使用 `MERGE()` hint，可关闭对当前子查询的物化过程，并将内部查询的内联展开到外部查询。该 hint 适用于非递归的公共表表达式查询，在某些场景下，使用该 hint 会比默认分配一块临时空间的语句执行效率更高。例如将外部查询的条件下推或在嵌套的 CTE 查询中： 
+
+{{< copyable "sql" >}}
+
+```sql
+-- 使用 hint 将外部查询条件的谓词下推
+WITH CTE AS (SELECT /*+ MERGE() */ * FROM tc WHERE tc.a < 60) SELECT * FROM CTE WHERE CTE.a <18;
```
Comment (Oreoxmt) [inline optimizer-hints.md:297]:
```suggestion
WITH CTE AS (SELECT /*+ MERGE() */ * FROM tc WHERE tc.a < 60) SELECT * FROM CTE WHERE CTE.a < 18;
```

File: optimizer-hints.md [hunk 17]
```diff
@@ -286,6 +286,29 @@ SHOW WARNINGS;
...
+
+-- 在嵌套 CTE 查询中使用该 hint 来指定将某个 CTE 内联展开到外部查询
+WITH CTE1 AS (SELECT * FROM t1), CTE2 AS (WITH CTE3 AS (SELECT /*+ MERGE() */ * FROM t2) ,CTE4 AS (SELECT * FROM t3) SELECT * FROM CTE3,CTE4) SELECT * FROM CTE1,CTE2;
```
Comment (Oreoxmt) [inline optimizer-hints.md:300]:
```suggestion
WITH CTE1 AS (SELECT * FROM t1), CTE2 AS (WITH CTE3 AS (SELECT /*+ MERGE() */ * FROM t2), CTE4 AS (SELECT * FROM t3) SELECT * FROM CTE3, CTE4) SELECT * FROM CTE1, CTE2;
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

**Pull request merged**